src/components/workMenuEntry.svelte

```diff
@@ -1,220 +1,150 @@
 <script lang="ts">
   import IconButton from '@src/components/iconButton.svelte';
-  import Input from '@src/components/input.svelte';
-  import { get, type Writable } from 'svelte/store';
-  import { saveResumeDataToLocalStorage, type Highlight } from '@src/data/data';
+  import {
+    saveResumeDataToLocalStorage,
+    type Work,
+    workStore,
+    type Highlight
+  } from '@src/data/data';
   import { arrayMove } from '@src/util/arrayMove';
-  import { onInput } from '@src/util/eventListeners';
   import { getDateValue } from '@src/util/getDateValue';
-  import { Tag, addTag, getTag } from '@src/data/tag';
+  import { get } from 'svelte/store';
+  import HighlightsSubMenu from './highlightsSubMenu.svelte';
 
+  /**
+   * The index of the work item. Required for operations on a specific work item.
+   */
   export let i: number;
-  export let visible: Writable<boolean>;
-  export let name: Writable<string>;
-  export let position: Writable<string>;
-  export let startDate: Writable<string>;
-  export let endDate: Writable<string>;
-  export let highlights: Writable<Array<Highlight>>;
+  export let visible: boolean;
+  export let name: string;
+  export let position: string;
+  export let startDate: string;
+  export let endDate: string;
+  export let highlights: Highlight[];
 
   const maxDate = getDateValue();
 
-  function reportValidity(e: Event) {
-    const target = e.target as HTMLInputElement;
-    if (target) {
-      target.reportValidity();
-    }
+  function updateWorkProperty(propname: keyof Work, value: string) {
+    workStore.update((workArray) => {
+      // I hate this method, and I wish I could just easily assign a property by array index
+      // but hey, this is easier to do
```

Comment:
If I could just do something like `workArray[i][propname]` without the compiler freaking out on me and without having to mess with `Record` types or other Interface magic, then that would be swell. However, this seems to work just fine for my needs.